backend/src/api/eam/eam.api.ts

```diff
@@ -93,6 +100,29 @@ const initEamApi: FastifyPluginAsync<Options> = async (fastify, opts) => {
     },
   });
 
+  fastify.route({
+    method: HttpMethod.GET,
+    url: EAMApiPath.ROOT,
+    async handler(req, rep) {
+      const [, token] = req.headers?.authorization?.split(' ') ?? [];
+
+      const user = await authService.getCurrentUser(token);
+      const hasPermission = await checkHasPermission(
+        [Permission.MANAGE_EAM],
+        user?.user.permissions ?? [],
+      );
+
+      if (!hasPermission) {
+        throw new EAMError({
+          status: HttpCode.DENIED,
+          message: ExceptionMessage.REQUIRED_PERMISSION,
+        });
+      }
```

Comment:
It is too much code for permission validation. After copying it to all other routes our controller file will be 2 times larger. Also it violates DRY

Reply:
agreed. just add a plugin